src/Packetery/Module/Checkout.php

```diff
@@ -1137,14 +1137,16 @@ private function isShippingRateRestrictedByProductsCategory( string $shippingRat
 		}
 
 		foreach ( $cartProducts as $cartProduct ) {
+			if ( ! isset( $cartProduct['product_id'] ) ) {
+				continue;
+			}
 			$product            = WC()->product_factory->get_product( $cartProduct['product_id'] );
 			$productCategoryIds = $product->get_category_ids();
 
 			foreach ( $productCategoryIds as $productCategoryId ) {
 				$productCategoryEntity           = ProductCategory\Entity::fromTermId( (int) $productCategoryId );
 				$disallowedCategoryShippingRates = $productCategoryEntity->getDisallowedShippingRates();
```

Comment:
Metodu getDisallowedShippingRates() prosím připodobnit k metodě Product::getDisallowedShippingRateIds(), která vrací jen to co je dané názvem metody.